src/markup.cpp

```diff
@@ -195,3 +197,70 @@ void markup_html(std::ostream &os, Expr *root) {
   JSONRender(os).render(root);
   os << "</script>" << std::endl;
 }
+
+void markup_ctags(std::ostream &os, Expr *root, const std::vector<std::pair<std::string, std::string>>& globals) {
+  for (auto &g : globals) {
+    Expr *e = root;
+    while (e && e->type == &DefBinding::type) {
+      DefBinding *d = static_cast<DefBinding*>(e);
+      e = d->body.get();
+      auto i = d->order.find(g.second);
+      if (i != d->order.end()) {
+        int idx = i->second.index;
+        Expr *v = idx < (int)d->val.size() ? d->val[idx].get() : d->fun[idx-d->val.size()].get();
+
```

Comment:
This algorithm is now copy-pasted into three places. It would probably be more DRY to transform 'globals' into a suitable form before handing it to these functions and/or the normal exports dumping algorithm.